File: PropertyManagementUI/src/app/Models/Building.ts
import { BaseModel } from "./BaseModel";
import { Community } from "./Community";

export interface Building extends BaseModel {
    Name: string | null;
    CommunityId: number;
    Community: Community | null;
    UseCommunityAddress: boolean | null;
    AddressLine1: string | null;
    AddressLine2: string | null;
    City: string | null;
    State: string | null;
    Country: string | null;
    Zipcode: string | null;
}

File: PropertyManagementSystem/PropertyManagementSystem/Models/Building.cs
using System.ComponentModel.DataAnnotations;

namespace PropertyManagementSystem.Models
{
    public class Building:BaseModel
    {
        [Required,StringLength(40)]
        public string Name { get; set; }
        public long CommunityId { get; set; }
        public Community Community { get; set; }
        public bool UseCommunityAddress { get; set; }
        public string AddressLine1 { get; set; }
        public string AddressLine2 { get; set; }

        [StringLength(50)]
        public string City { get; set; }

        [StringLength(50)]
        public string State { get; set; }

        [StringLength(60)]
        public string Country { get; set; }

        [StringLength(10)]
        public string Zipcode { get; set; }
    }
}
